hack/go-linter.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+set -e
+
+source ./hack/common.sh
+
+fetch_go_linter
+
+header_text "Running golangci-lint"
+golangci-lint run --disable-all \
+    --deadline 5m \
+    --enable=nakedret \
+
+##todo(camilamacedo86): The following checks requires fixes in the code
+# --enable=golint
+# --enable=gocyclo
+# --enable=goimports
+# --enable=lll
+# --enable=goconst
+# --enable=gosec
+# --enable=maligned
+# --enable=deadcode \
+# --enable=misspell \
+# --enable=errcheck \
+# --enable=dupl \
+# --enable=interfacer \
+# --enable=misspell \
+# --enable=ineffassign \
+# --enable=varcheck \
+# --enable=structcheck \
+# --enable=unparam \
```

Comment:
Now we can create one issue for each kind. 
So, in each issue we should enable the check and solve the problems.
c/c @joelanford @jmrodri @hasbro17